src/components/fileStorage/repo/files.repo.js

```diff
@@ -0,0 +1,41 @@
+const { FileModel } = require('./db');
+
+/**
+ * @param {*} fileIds
+ * @param {*} userId
+ * @return {MongooseBatchResult}
+ */
+const removeFilePermissionsByUserId = async (userId) => {
+	const searchQuery = { permissions: { $elemMatch: { refId: userId } } };
+
+	const filePermissions = await FileModel.aggregate([
+		{
+			$match: {
```

Comment:
$match vs $elemMatch

Reply:
$match is an aggregation stage, $elemMatch is a query operator